sklearn/tests/test_common.py

```diff
@@ -231,3 +233,27 @@ def test_all_tests_are_importable():
                                  '__init__.py or an add_subpackage directive '
                                  'in the parent '
                                  'setup.py'.format(missing_tests))
+
+
+WHITELIST = set(['n_components', 'estimator', 'base_estimator', 'n_clusters',
+                 'n_neighbors', 'steps', 'regressor', 'transformers',
+                 'n_estimators', 'transformer_list', 'score_func', 'func',
+                 'degree', 'estimators'])
+
+
+@pytest.mark.parametrize(
+    "name, Estimator",
+    [(name, Estimator)
+     for name, Estimator in all_estimators() if not name.startswith("_")])
+def test_estimators_keyword_only(name, Estimator):
```

Comment:
This test makes sure that the whitelist is POSITION_OR_KEYWORD and that the rest of the parameters (without defaults) is KEYWORD_ONLY.